src/main/scala/edu/berkeley/cs/rise/opaque/benchmark/TPCH.scala

```diff
@@ -138,59 +141,71 @@ object TPCH {
       .option("delimiter", "|")
       .load(s"${Benchmark.dataDir}/tpch/$size/nation.tbl")
       .repartition(numPartitions))
+}
 
+class TPCH(
+  val sqlContext: SQLContext,
+  val size: String,
+  val numPartitions: Int,
 
-  private def tpch9EncryptedDFs(
-      sqlContext: SQLContext, securityLevel: SecurityLevel, size: String, numPartitions: Int)
-       : (DataFrame, DataFrame, DataFrame, DataFrame, DataFrame, DataFrame) = {
-      val partDF = part(sqlContext, securityLevel, size, numPartitions)
-      val supplierDF = supplier(sqlContext, securityLevel, size, numPartitions)
-      val lineitemDF = lineitem(sqlContext, securityLevel, size, numPartitions)
-      val partsuppDF = partsupp(sqlContext, securityLevel, size, numPartitions)
-      val ordersDF = orders(sqlContext, securityLevel, size, numPartitions)
-      val nationDF = nation(sqlContext, securityLevel, size, numPartitions)
-      (partDF, supplierDF, lineitemDF, partsuppDF, ordersDF, nationDF)
+  val nameToDF: Map[String, DataFrame],
+  val nameToEncryptedDF: Map[String, DataFrame],
+) {
+
+  val tableNames = Seq("part", "supplier", "lineitem", "partsupp", "orders", "nation")
+
+  def this(
+    sqlContext: SQLContext,
+    size: String,
+    numPartitions: Int,
+  ) = {
+    this(sqlContext, size, numPartitions, 
+
+    Map("part" -> TPCHDataFrames.part(sqlContext, Insecure, size, numPartitions),
+    "supplier" -> TPCHDataFrames.supplier(sqlContext, Insecure, size, numPartitions),
+    "lineitem" -> TPCHDataFrames.lineitem(sqlContext, Insecure, size, numPartitions),
+    "partsupp" -> TPCHDataFrames.partsupp(sqlContext, Insecure, size, numPartitions),
+    "orders" -> TPCHDataFrames.orders(sqlContext, Insecure, size, numPartitions),
+    "nation" -> TPCHDataFrames.nation(sqlContext, Insecure, size, numPartitions)),
+
+    Map("part" -> TPCHDataFrames.part(sqlContext, Encrypted, size, numPartitions),
+    "supplier" -> TPCHDataFrames.supplier(sqlContext, Encrypted, size, numPartitions),
+    "lineitem" -> TPCHDataFrames.lineitem(sqlContext, Encrypted, size, numPartitions),
+    "partsupp" -> TPCHDataFrames.partsupp(sqlContext, Encrypted, size, numPartitions),
+    "orders" -> TPCHDataFrames.orders(sqlContext, Encrypted, size, numPartitions),
+    "nation" -> TPCHDataFrames.nation(sqlContext, Encrypted, size, numPartitions)),
+    )
+    ensureCached()
+  }
```

Comment:
This constructor setup is currently a bit confusing to read. Is it possible to put this in a separate setup function, which can then be called by TPCHTests before the tests are run?